### PR TITLE
feat(CreateChatView): Support pasting chat key

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/7176

### What does the PR do

Open a "Send contact request" popup when user pastes a valid chat key to "create chat" text field.
Supported values are:
- Public key
- Public key wrapped in a URL
- Compressed key
- Compressed key wrapped in a URL

⚠️  I also made an important (but not a breaking) change - I switched `Global` root item from `QtObject` to `Item`, so that Components could be create right there. This is needed so that we could get the reference to created popup whenever we use `Global.openContactRequestPopup`. In this particular case we bind to it's `closed` signal.

⚠️ Also I made changes to Nim contacts service:
`getContactById` now supports compressed keys.

### Affected areas

CreateChat, ProfilePopup

### Screenshot of functionality (including design for comparison)

https://user-images.githubusercontent.com/25482501/190861714-7213eeb0-42bc-41bd-8d9c-3b7be2e9726b.mov

### CODE REVIEW

File: ui/app/AppLayouts/Chat/panels/InlineSelectorPanel.qml [hunk 1]
```diff
@@ -122,6 +129,17 @@ Item {
                                 }
 
                                 Keys.onPressed: {
+
+                                    if ((event.modifiers | Qt.ControlModifier) && event.key === Qt.Key_V) {
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/InlineSelectorPanel.qml:133]: Better (also) check for `event.matches(StandardKey.Paste)`

File: ui/app/AppLayouts/Chat/views/CreateChatView.qml [hunk 2]
```diff
@@ -48,7 +48,7 @@ Page {
                 Layout.rightMargin: 16
 
                 rootStore: root.rootStore
-                enabled: root.rootStore.contactsModel.count > 0
+//                enabled: root.rootStore.contactsModel.count > 0
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/CreateChatView.qml:51]: Why commented out?

File: ui/app/AppLayouts/Chat/views/MembersSelectorView.qml [hunk 3]
```diff
@@ -48,12 +54,25 @@ MembersSelectorBase {
 
         property ListModel selectedMembers: ListModel {}
 
+        property var lookupContact: Backpressure.debounce(root, 0, function (value) {
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/MembersSelectorView.qml:57]: Why not a simple `function()`?

File: ui/imports/shared/popups/qmldir [hunk 4]
```diff
@@ -22,3 +22,4 @@ ImageCropWorkflow 1.0 ImageCropWorkflow.qml
 ImportCommunityPopup 1.0 ImportCommunityPopup.qml
 DisplayNamePopup 1.0 DisplayNamePopup.qml
 AddEditSavedAddressPopup 1.0 AddEditSavedAddressPopup.qml
+SendContactRequestModal 1.0 SendContactRequestModal.qml
```
Comment (caybro) [inline ui/imports/shared/popups/qmldir:25]: Again, this file seems to be (physically) duplicated in 2 different subdirectories; please remove one

**Review by caybro — CHANGES_REQUESTED** [review]
Generally looks good, just some minor issues

File: ui/imports/utils/Global.qml [hunk 5]
```diff
@@ -41,6 +43,16 @@ QtObject {
     signal openEditDisplayNamePopup()
     signal openActivityCenterPopupRequested
 
+    function openContactRequestPopup(publicKey) {
+        const store = appMain.rootStore.profileSectionStore.contactsStore
```
Comment (osmaczko) [inline ui/imports/utils/Global.qml:47]: unused

File: ui/imports/utils/Global.qml [hunk 6]
```diff
@@ -107,4 +119,14 @@ QtObject {
     function settingsHasLoaded() {
         settingsLoaded()
     }
+
+    Component {
+        id: sendContactRequestPopupComponent
+        SendContactRequestModal {
+            anchors.centerIn: parent
+            header.title: qsTr("Send Contact Request to %1").arg(userDisplayName)
```
Comment (osmaczko) [inline ui/imports/utils/Global.qml:127]: Shouldn't it be a part of `SendContactRequestModal` :thinking: ?

File: ui/imports/utils/Global.qml [hunk 7]
```diff
@@ -41,6 +43,16 @@ QtObject {
...
     signal openActivityCenterPopupRequested
 
+    function openContactRequestPopup(publicKey) {
+        const store = appMain.rootStore.profileSectionStore.contactsStore
+        const contactDetails = Utils.getContactDetailsAsJson(publicKey);
+        return openPopup(sendContactRequestPopupComponent, {
+            userPublicKey: publicKey,
+            userDisplayName: contactDetails.displayName,
+            userIcon: contactDetails.largeImage,
+        })
```
Comment (osmaczko) [inline ui/imports/utils/Global.qml:53]: This is inconsistent with how popups are already opened through `Global.qml`. 

```suggestion
        openSendContactRequestRequested(publicKey, contactDetails.displayName, contactDetails.largeImage)
```

**Review by osmaczko — CHANGES_REQUESTED** [review]
Functionally looks good, some comments left.

File: ui/app/AppLayouts/Chat/views/MembersSelectorView.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Chat/views/MembersSelectorView.qml:57]: I guess they would effectively do the same; just like a `QTimer::singleShot()` or `QMetaObject::invokeMethod()`; ie. it forces processing of events

**Comment by glitchminer** [comment]
## Issues

#### 1. OPEN [Observation] What should happen when after the contact request has been sent
Currently the chat key is just left in the input field. The user must manually clear it to chat with someone else.

Discussed option of closing the Start Chat screen all together - @igor-sirotin to discuss with @John-44  

#### 2. OPEN [Observation] It's not possible to type the key to add the contact

#### 3. MOVED Confirm button is enabled when field is empty and when text is entered but not a contact

#### 4. MOVED Not possible to right click on field to paste

#### 5. MOVED After entering key, the contact list is filtered but there is no indication that there are just no results to select

**Review by glitchminer — APPROVED** [review]
Tested, approved.

**Review by caybro — APPROVED** [review]

**Pull request merged**